src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -347,6 +347,106 @@ internal static DkmClrValue GetNullableValue(this DkmClrValue value, DkmInspecti
             return value.GetFieldValue(InternalWellKnownMemberNames.NullableValue, inspectionContext);
         }
 
+        internal static void GetTupleFieldTypes(this Type type, ArrayBuilder<Type> fieldTypes)
+        {
+            while (true)
+            {
+                var typeArguments = type.GetGenericArguments();
+                int n = typeArguments.Length;
+                Debug.Assert(n > 0);
+                Debug.Assert(n <= TupleFieldRestPosition);
+                if (n < TupleFieldRestPosition)
+                {
+                    fieldTypes.AddRange(typeArguments);
+                    return;
+                }
+                else
+                {
+                    for (int i = 0; i < n - 1; i++)
+                    {
+                        fieldTypes.Add(typeArguments[i]);
```

Comment:
Why do we need a local variable for ```typeArguments.Length``` at all? It makes the code unnecessarily complicated.